watchers/configmap_watcher.go

```diff
@@ -68,21 +67,21 @@ func checkConfigMapChanges(eventChannel <-chan watch.Event, dataField string, da
 	}
 }
 
-func createWatcher(clientset kubernetes.Interface, namespace string, configMapName string, watcher *watch.Interface) error {
-	err := retry.OnError(backoff, retriable, func() error {
-		var innerErr error
-		*watcher, innerErr = clientset.CoreV1().ConfigMaps(namespace).Watch(context.TODO(),
-			metav1.SingleObject(metav1.ObjectMeta{Name: configMapName, Namespace: namespace}))
-		if innerErr != nil {
-			logger.Info("cannot create watcher", "namespace", namespace, "configMap name", configMapName, "error", innerErr)
-		}
-		return innerErr
-	})
-	return err
+// watcher implements config.Watch interface which will create a ConfigMapWatcher.
+type watcher struct {
```

Comment:
WDYT to move this struct declaration below the `var` section? I think this can make the code style a bit more consistent.

Reply:
ok but this is private. I think the code is more clear with this structure here.
I don't have a strong opinion about it. 
I let you decide :)

Reply:
No strong opinion. Please do whatever you prefer :)

Reply:
I think it should be moved to the top; structs with structs, methods later.